yt_dlp/extractor/nobelprize.py

```diff
@@ -1,59 +1,62 @@
 from .common import InfoExtractor
 from ..utils import (
-    determine_ext,
-    get_element_by_attribute,
-    int_or_none,
-    js_to_json,
-    mimetype2ext,
-    update_url_query,
+    clean_html,
+    parse_duration,
+    parse_qs,
+    unified_timestamp,
+    update_url,
+    url_or_none,
+)
+from ..utils.traversal import (
+    find_element,
+    require,
+    traverse_obj,
 )
 
 
 class NobelPrizeIE(InfoExtractor):
-    _WORKING = False
-    _VALID_URL = r'https?://(?:www\.)?nobelprize\.org/mediaplayer.*?\bid=(?P<id>\d+)'
-    _TEST = {
-        'url': 'http://www.nobelprize.org/mediaplayer/?id=2636',
-        'md5': '04c81e5714bb36cc4e2232fee1d8157f',
+    _VALID_URL = r'https?://(?:(?:mediaplayer|www)\.)?nobelprize\.org/mediaplayer/'
+    _TESTS = [{
+        'url': 'https://www.nobelprize.org/mediaplayer/?id=2636',
         'info_dict': {
             'id': '2636',
             'ext': 'mp4',
             'title': 'Announcement of the 2016 Nobel Prize in Physics',
-            'description': 'md5:05beba57f4f5a4bbd4cf2ef28fcff739',
+            'description': 'md5:1a2d8a6ca80c88fb3b9a326e0b0e8e43',
+            'duration': 1560.0,
+            'thumbnail': r're:https?://www\.nobelprize\.org/images/.+\.jpg',
+            'timestamp': 1504883793,
+            'upload_date': '20170908',
         },
-    }
+    }, {
+        'url': 'https://mediaplayer.nobelprize.org/mediaplayer/?qid=12693',
+        'info_dict': {
+            'id': '12693',
+            'ext': 'mp4',
+            'title': 'Nobel Lecture by Peter Higgs',
+            'description': 'md5:9b12e275dbe3a8138484e70e00673a05',
+            'duration': 1800.0,
+            'thumbnail': r're:https?://www\.nobelprize\.org/images/.+\.jpg',
+            'timestamp': 1504883793,
+            'upload_date': '20170908',
+        },
+    }]
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
+        url = update_url(url, netloc='mediaplayer.nobelprize.org')
+        video_id = traverse_obj(parse_qs(url), (
+            ('id', 'qid'), 0, any, {require('video ID')}))
         webpage = self._download_webpage(url, video_id)
```

Comment:
I'd do this to maintain the old behavior
```suggestion
        video_id = traverse_obj(parse_qs(url), (('id', 'qid'), -1, {int_or_none}, {str_or_none}, any))
        if not video_id:
            raise UnsupportedError(url)
        webpage = self._download_webpage(
            update_url(url, netloc='mediaplayer.nobelprize.org'), video_id)
```